File: src/my_py_pkg/my_py_pkg/add_two_ints_client_no_oop.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts

def main(args=None):
    rclpy.init(args=args)
    # criando um objeto Node chamado add_two_ints_no_oop
    node = Node("add_two_ints_no_oop") 
    # criando o cliente
    client =  node.create_client(AddTwoInts, "add_two_ints")

    # enquanto o servidor nao responde
    while not client.wait_for_service(1.0):
        node.get_logger().warn("Waiting for Server Add Two Ints")

    # criando um objeto de request
    request =  AddTwoInts.Request()
    request.a = 3
    request.b = 8

    # manda o request de forma assicrona para a variavel future q recebera a resposta
    future = client.call_async(request)
    # deixa o no rodando ate ter a resposta
    rclpy.spin_until_future_complete(node, future)

    try:  # recebe a resposta do service e print a resposta
        response = future.result()
        node.get_logger().info(str(request.a) + " + " + str(request.b) + " = " + str(response.sum))
        
    except Exception as e:
        node.get_logger().error("Service call failed %r" % (e,))
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/add_two_ints_client_no_oop.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

    
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("add_two_ints_client_no_oop"); 

    // criando um client
    auto client = node->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
    // mandando uma mensagem enquanto o server nao responde
    while(!client->wait_for_service(std::chrono::seconds(1))){
        RCLCPP_WARN(node->get_logger(),  "Waiting for the server to be up...");
    }

    // criando request e passando os numeros
    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 3;
    request->b = 8;

    // resposta recebida do server
    auto future = client->async_send_request(request);

    // printando o resultado da interação client/server
    if (rclcpp::spin_until_future_complete(node, future) == rclcpp::FutureReturnCode::SUCCESS)
    {
        RCLCPP_INFO(node->get_logger(), "%ld + %ld = %d", request->a, request->b, (int)future.get()->sum);
    }
    else
    {
        RCLCPP_ERROR(node->get_logger(), "Error while calling service");
    }
    

    rclcpp::shutdown();
    return 0;
}